### PR TITLE
Integration spec

### PR DESCRIPTION
**Pull request by kaizipaul**
### In this PR, I made the following changes:
- Solved the n+1 problem:
### Before n+1:
posts_before_n+1
users_before_n+1

### After n+1:
posts_after_n+1
users_after_n+1

> Wrote integration tests for:
- Post index page.
- Post show page.
- Users index page.
- Users show page.

### CODE REVIEW

File: spec/integration/user_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  it 'shows the photo of each user' do
+    assert page.has_xpath?("//img[@src = 'https://unsplash.com/photos/F_-0BxGuVvo'
+    and @alt='user photo']")
+    assert page.has_xpath?("//img[@src = 'https://unsplash.com/photos/i2hoD-C2RUA'
+    and @alt = 'user photo']")
+  end
+
+  it 'shows the users posts' do
+    expect(page).to have_content('Number of posts: 3')
+  end
```
Comment (TadesseDev) [inline spec/integration/user_index_spec.rb]: - [ ] Great job @kaizipaul, Your tests are clean and sweet, Yet according to the requirements we need a test that asserts you can see the number of posts `each user` has written. I can see you check for the first user, please add a test case for the second user to make sure your test is accurate all the way.

File: spec/integration/post_index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  end
+
+  it 'shows number of likes for a post' do
+    expect(page).to have_content('Likes: 0')
+  end
+
+  it 'redirects to posts show page' do
+    click_on 'Hello'
+    expect(page).to have_current_path user_post_path(@kyle.id, @post.id)
+  end
```
Comment (TadesseDev) [inline spec/integration/post_index_spec.rb]: - [ ] You did a great job @kaizipaul, But here you have missed a test that should be part of this test suit. We need to add a test that asserts you can see a section for pagination if there are more posts than fit on the view. This asserts that pagination is always visible to users. Please add this one.

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi @kaizipaul 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
212094426-b41dd457-c9db-4ab3-bbe9-176dd8dd08b7


## Highlights
 - [ ]  Correct GitFlow  ✅
 - [ ]  Good commit history  ✅
 - [ ]  Good PR title/description  ✅
 - [ ]  All linters passed  ✅
 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by t-yanick — APPROVED** [review]
# STATUS: MILESTONE APPROVED 

Hi @kaizipaul , 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  All changes requested by previous reviewer implemented
:heavy_check_mark: Before and after screenshots of solving N+1 problem added in the PR description 
:heavy_check_mark:  All test cases added
:heavy_check_mark: All linter checks passing 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**